### DISCUSSION TITLE
diary 主题兼容问题

### DISCUSSION THREAD

**Discussion by tsq456** [walinejs/waline#168] [Q&A] [answered]
**问题描述 | Describe the bug**
部署在博客后，评论框下方的内容区域一直处于刷新状态，就是转圈圈，然后页没法发评论，但是我在vercel部署后的地址是可以发布评论的。

**问题网站 | Website URL**
atugo.me

**部署方式 | Deploy Type**
- 服务部署在哪里？ | Where your waline deploy?
    - [ x ] Vercel

- 数据存储在哪里？| Where your comment data store?
    - [ x ] LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment]
serverURL 的配置去掉最后的 `/` 即可，`https://blogcomment-git-master-tsq456.vercel.app`。

**Comment by tsq456** [comment]
问题还是会出现，已经把 / 去掉了

**Comment by lizheming** [comment]
如果你的博客仓库是公开的话能否提供一下你的博客仓库，或者是你使用的主题地址。我本地没有复现这个问题，准备看一下是不是主题兼容问题。

**Comment by tsq456** [comment]
主题地址是：https://github.com/amazingrise/hugo-theme-diary
博客仓库是：https://github.com/tsq456/aTuGo.git

用的是hugo，感谢帮助^^

**Comment by tsq456** [comment]
我自己测试了下，估计是主题兼容性的问题，目前这个主题用twikoo也会出现类似的情况_（点击评论按钮没有反应）_，用别人写的豆瓣观影记录也会出现无法加载的问题。想问下情况一般是由什么引起的呢？自己是否有办法排查

**Comment by tsq456** [comment]
补充下，我把waline接到hugo其他主题下，是可行的

**Comment by lizheming** [comment]
不要急，晚点我看一下

**Comment by lizheming** [comment, marked as answer]
https://github.com/tsq456/aTuGo/blob/master/aTuGo/themes/diary/layouts/partials/journal.html
我看了下应该就是主题的兼容性问题了，这个主题整体用了一层 Vue 框架重新渲染了页面，而 Waline 和 Twikoo 都是使用 React/Vue 这种前端框架渲染的，外部重新监听 DOM 会导致渲染发生异常。
解决的办法有两种，要么是将主题的 Vue 渲染去除掉使用原生 JS 处理。要么就是把评论的 HTML 放在 `<div id="app">` 外面，这样就不会有冲突的问题了。有能力的话我建议选择前者处理一下，看了下代码似乎没必要大费周章的用 Vue 去做处理。

**Comment by tsq456** [comment]
明白了，谢谢啦！我看看怎么处理下
